test/integration-serial/resource_discovery_test.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+require 'test_helper'
+
+class ResourceDiscoveryTest < KubernetesDeploy::IntegrationTest
+  def test_non_prunable_crd
+    assert_deploy_success(deploy_fixtures("resource-discovery", subset: ["shredders.yml"]))
+    assert_deploy_success(deploy_fixtures("resource-discovery", subset: ["shredders_cr.yml"]))
```

Comment:
Are these two separate deploys because CRDs aren't in the priority list?

Reply:
No, the CRD has to be added to the cluster before the CR can pass validation. Since we do validation before we deploy even pre-deploy resources we need to deployments.

Reply:
Ah, that makes sense.